Google.PowerShell/Common/GcloudCmdlet.cs

```diff
@@ -87,9 +88,31 @@ protected string GetFullPath(string filePath)
                 return filePath;
             }
 
-            // For relative files, qualify them based on where the PowerShell session is.
-            string currentDirectory = SessionState.Path.CurrentLocation.Path;
-            return Path.Combine(currentDirectory, filePath);
+            // Try to resolve the path using PowerShell (only applicable for FileSystemProvider).
+            try
+            {
+                ProviderInfo provider = null;
+                string[] result = GetResolvedProviderPathFromPSPath(filePath, out provider).ToArray();
+
+                // Only return the resolved path if there are no ambiguities.
+                // If path contains wildcards, then it may resolved to more than 1 path.
+                if (result.Length == 1 && provider.ImplementingType == typeof(FileSystemProvider))
```

Comment:
Let's say results is null, then this crashes. If `result?.Length ==1 ` compiles, why don't we go with that.